setuptools/config/_apply_pyprojecttoml.py

```diff
@@ -153,15 +152,18 @@ def _guess_content_type(file: str) -> Optional[str]:
     raise ValueError(f"Undefined content type for {file}, {msg}")
 
 
-def _long_description(dist: "Distribution", val: _DictOrStr, root_dir: StrPath):
+def _long_description(
+    dist: "Distribution", val: _ProjectReadmeValue, root_dir: StrPath
+):
     from setuptools.config import expand
 
+    file: Union[str, tuple[()]]
     if isinstance(val, str):
-        file: Union[str, list] = val
+        file = val
         text = expand.read_files(file, root_dir)
-        ctype = _guess_content_type(val)
+        ctype = _guess_content_type(file)
     else:
-        file = val.get("file") or []
+        file = val.get("file", ())
```

Comment:
This is better right? (e.g. `{"file": ""}`)

```suggestion
        file = val.get("file") or ()
```

Reply:
Ah oops. Yes, didn't double check against the original implementation before my PR.